Core/Core/Configuration/ServerConfig/ServerConfig.swift

```diff
@@ -0,0 +1,61 @@
+//
+//  ServerConfig.swift
+//  Core
+//
+//  Created by Saeed Bashir on 4/29/24.
+//
+
+import Foundation
+
+private enum Keys: String, RawStringExtractable {
+    case valuePropEnabled = "value_prop_enabled"
+}
+
+public protocol ServerConfigProtocol {
+    var valuePropEnabled: Bool { get }
+    var iapConfig: IAPConfig { get }
+    func initialize(serverConfig: String?)
+}
+
+public class ServerConfig: ServerConfigProtocol {
+    var config: [String: Any] = [:]
+    
+    public init () {}
+    
+    public func initialize(serverConfig: String?) {
+        guard let serverConfig = serverConfig,
+                let configData = serverConfig.data(using: .utf8),
+              let config = try? JSONSerialization.jsonObject(with: configData, options: []) as? [String: Any]
+        else { return }
+        
+        self.config = config
+    }
+    
+    public var valuePropEnabled: Bool {
+        return config[Keys.valuePropEnabled] as? Bool ?? false
+    }
+}
+
+// Mark - For testing and SwiftUI preview
+#if DEBUG
+public class ServerConfigProtocolMock: ServerConfigProtocol {
+    
+    let configString = "{\"iap_config\":{\"enabled\":true,\"experiment_enabled\":false,\"android_product_prefix\":\"mobile.android.usd\",\"allowed_users\":[\"all_users\"]},\"value_prop_enabled\":true,\"feedback_form_url\":\"https://bit.ly/edx-apps-feedback\",\"course_dates_calendar_sync\":{\"ios\":{\"enabled\":true,\"self_paced_enabled\":true,\"instructor_paced_enabled\":true,\"deep_links_enabled\":true},\"android\":{\"enabled\":true,\"self_paced_enabled\":true,\"instructor_paced_enabled\":true,\"deep_links_enabled\":true}}}"
+    
```

Comment:
Alternate approach could be to read it from mock file may be. 🤔

Reply:
Using the same approach for config, so we can live with it.